IIBB: fix old Python2 code

abernardi opened the pull request:
Fix old python2 code in module to get ARBA data

abernardi committed:
IIBB: fix old Python2 code

reingart commented on iibb.py:
```diff
@@ -122,29 +123,27 @@ def ConsultarContribuyentes(self, fecha_desde, fecha_hasta, cuit_contribuyente):
+                        user=self.Usuario, password=self.Password, file=archivo)
             else:
-                response = open(self.testing).read()
+                with open(self.testing).read() as archivo:
```
Para el context manager, el `read` debería estar en el archivo, ¿o lo armaron asi por algún motivo en especial?:
```suggestion
                with open(self.testing) as archivo:
```

reingart commented on iibb.py:
```diff
@@ -122,29 +123,27 @@ def ConsultarContribuyentes(self, fecha_desde, fecha_hasta, cuit_contribuyente):
             else:
-                response = open(self.testing).read()
+                with open(self.testing).read() as archivo:
+                    response = archivo
```
```suggestion
                    response = archivo.read()
```

reingart commented on iibb.py:
```diff
@@ -22,7 +22,8 @@
 
-import md5, os, sys, tempfile, traceback
+import os, sys, tempfile, traceback
+from hashlib import md5
```
Aca se podría soprtar python 2 y 3 capturando la excepción:
```suggestion
try:
    from hashlib import md5
except ImportError:
    from md5 import md5
```

reingart reviewed (approved):
Gracias @abernardi por la contribución!
Deje unos comentarios para ajustes menores y si se puede soportar tambien python 2 (pero si se complica no hay problemas, podemos dejar de soportarlo).

abernardi committed:
IIBB: fix old Python2 code (2)

abernardi commented:
Hola @reingart. Gracias por las observaciones, eran errores de código.
Agregué otro commit con los ajustes para soportar ambas versiones de python.
Saludos!

reingart committed:
IIBB: Fix hash md5 unicode encoding error

```
Traceback: Traceback (most recent call last):
  File "/src/abernardi/iibb.py", line 129, in ConsultarContribuyentes
    self.CodigoHash = md5(xml.decode('utf-8')).hexdigest()
TypeError: Strings must be encoded before hashing
```

.

reingart committed:
IIBB: fix temp file unicode error (binary open)

```
Traceback: Traceback (most recent call last):
  File "/src/abernardi/iibb.py", line 134, in ConsultarContribuyentes
    archivo.write(xml)
TypeError: write() argument must be str, not bytes
```

reingart committed:
IIBB: fix error message unicode encoding

Note b prefix in the error message and wrong characters (?):
```
Numero Comprobante:
Codigo HASH: f1800fba55a255c8a6317b324bf5efc7
Error General: DATO | 2 | b'<![CDATA[El usuario ingresado y / o la contrase?a son inv?lidos.]]/>'
```

Now:
```
Numero Comprobante:
Codigo HASH: f1800fba55a255c8a6317b324bf5efc7
Error General: DATO | 2 | <![CDATA[El usuario ingresado y / o la contraseña son inválidos.]]/>
```

reingart commented:
Gracias nuevamente @abernardi por los ajustes!

Corregi algunas cositas más de unicode (hash, escritura archivo temporal y mensajes de error), en los commits deje los comentarios.

Lo probé por linea de comando y estaría funcionando OK tanto en python2 como python3 (al menos lo básico, calculo que con un usuario y clave productiva debería andar):

`python2 iibb.py 20267565393 CIT 20150301 20150331 27269434894 --trace`:
```
--------------------------------------------------------------------------------
POST https://dfe.test.arba.gov.ar/DomicilioElectronico/SeguridadCliente/dfeServicioConsulta.do
Content-length: 783
Content-type: multipart/form-data; boundary=127.0.1.1.1000.489.1693150912.913.1

--127.0.1.1.1000.489.1693150912.913.1
Content-Disposition: form-data; name="password"

CIT
--127.0.1.1.1000.489.1693150912.913.1
Content-Disposition: form-data; name="user"

20267565393
--127.0.1.1.1000.489.1693150912.913.1
Content-Disposition: form-data; name="file"; filename="DFEServicioConsulta_f1800fba55a255c8a6317b324bf5efc7.xml"
Content-Type: application/xml

<?xml version="1.0" encoding="UTF-8"?><CONSULTA-ALICUOTA>
    <cantidadContribuyentes>1</cantidadContribuyentes>
      <contribuyentes class="list">
        <contribuyente>
        <cuitContribuyente>27269434894</cuitContribuyente></contribuyente>
      </contribuyentes>
<fechaDesde>20150301</fechaDesde><fechaHasta>20150331</fechaHasta></CONSULTA-ALICUOTA>
--127.0.1.1.1000.489.1693150912.913.1--

status: 200
content-disposition: attachment; filename="DFERespuesta.xml"
content-language: en-US
x-powered-by: Servlet/3.1
transfer-encoding: chunked
set-cookie: JSESSIONID=0000A1aPMxOHLzsvI5kujSzaXS1:-1; Path=/; HttpOnly
expires: Thu, 01 Dec 1994 16:00:00 GMT
cache-control: no-cache="set-cookie, set-cookie2"
date: Sun, 27 Aug 2023 15:41:54 GMT
content-type: text/xml;charset=ISO-8859-1
<?xml version='1.0' encoding='ISO-8859-1'?><DFEError>
  <tipoError>DATO</tipoError>
  <codigoError>2</codigoError>
  <mensajeError>&lt;![CDATA[El usuario ingresado y / o la contrase�a son inv�lidos.]]/&gt;</mensajeError>
</DFEError>
================================================================================
Numero Comprobante:
Codigo HASH: f1800fba55a255c8a6317b324bf5efc7
Error General: DATO | 2 | <![CDATA[El usuario ingresado y / o la contraseña son inválidos.]]/>
```

`python3 iibb.py 20267565393 CIT 20150301 20150331 27269434894 --trace`:
```
--------------------------------------------------------------------------------
POST https://dfe.test.arba.gov.ar/DomicilioElectronico/SeguridadCliente/dfeServicioConsulta.do
Content-type: multipart/form-data; boundary================1109104537687450579==
Content-length: 787

--===============1109104537687450579==
Content-Disposition: form-data; name="user"

20267565393
--===============1109104537687450579==
Content-Disposition: form-data; name="password"

CIT
--===============1109104537687450579==
Content-Disposition: form-data; name="file"; filename="DFEServicioConsulta_f1800fba55a255c8a6317b324bf5efc7.xml"
Content-Type: application/xml

<?xml version="1.0" encoding="UTF-8"?><CONSULTA-ALICUOTA>
    <cantidadContribuyentes>1</cantidadContribuyentes>
      <contribuyentes class="list">
        <contribuyente>
        <cuitContribuyente>27269434894</cuitContribuyente></contribuyente>
      </contribuyentes>
<fechaDesde>20150301</fechaDesde><fechaHasta>20150331</fechaHasta></CONSULTA-ALICUOTA>
--===============1109104537687450579==--

x-powered-by: Servlet/3.1
content-type: text/xml;charset=ISO-8859-1
content-disposition: attachment; filename="DFERespuesta.xml"
content-language: en-US
set-cookie: JSESSIONID=0000wpGYsOv2oK1Wz87sIEbw8Ri:-1; Path=/; HttpOnly
transfer-encoding: chunked
date: Sun, 27 Aug 2023 15:41:56 GMT
expires: Thu, 01 Dec 1994 16:00:00 GMT
cache-control: no-cache="set-cookie, set-cookie2"
status: 200
b"<?xml version='1.0' encoding='ISO-8859-1'?><DFEError>\n  <tipoError>DATO</tipoError>\n  <codigoError>2</codigoError>\n  <mensajeError>&lt;![CDATA[El usuario ingresado y / o la contrase\xf1a son inv\xe1lidos.]]/&gt;</mensajeError>\n</DFEError>"
================================================================================
Numero Comprobante:
Codigo HASH: f1800fba55a255c8a6317b324bf5efc7
Error General: DATO | 2 | <![CDATA[El usuario ingresado y / o la contraseña son inválidos.]]/>
```
